Gems/Atom/RHI/Vulkan/Code/Source/RHI/Device.cpp

```diff
@@ -536,10 +550,15 @@ namespace AZ
             m_imageMemoryRequirementsCache.Clear();
             m_bufferMemoryRequirementsCache.Clear();
 
-            if ( m_nativeDevice != VK_NULL_HANDLE )
+            // Only destroy VkDevice if created locally and not passed in by a XR module
+            bool isXrDeviceActive = RHI::RHISystemInterface::Get()->GetXRSystem();
```

Comment:
minor: Odd implicit conversion between `RHI::XRRenderingInterface*` to bool. It'd be better to be explicit about what the function returns and do a comparison:

`bool isXrDeviceActive = RHI::RHISystemInterface::Get()->GetXRSystem() != nullptr;`